addons/account/models/account_move.py

```diff
@@ -314,6 +333,9 @@ def post(self):
         # Create the analytic lines in batch is faster as it leads to less cache invalidation.
         self.mapped('line_ids').create_analytic_lines()
         for move in self:
+            if move.auto_post and move.date > fields.Date.today():
+                raise UserError(_("This move is configured to be auto-posted. If you are sure you want to post it, please untick the checkbox."))
```

Comment:
raise UserError(_("This move is configured to be auto-posted on %s ...) % (move.date,)

Reply:
I feel like the code is ugly for the locale format but I didn't find anything else though